src/Compilers/CSharp/Portable/Binder/AwaitableInfo.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Internal structure containing all semantic information about an await expression.
+    /// </summary>
+    internal struct AwaitableInfo : IEquatable<AwaitableInfo>
```

Comment:
I think this could be a class.  (like `ForEachEnumeratorInfo`)

Making it a struct just adds complexity - dealing with nullables and such.

Reply:
And it will not save much in terms of allocations since these infos are not transient. They are always stored in the nodes thus occupying heap space anyways. 
There won't be a lot of them either.